source/material_model/rheology/visco_plastic.cc

```diff
@@ -729,7 +740,9 @@ namespace aspect
                                                                           n_phases_per_composition);
                 plastic_out->cohesions[i]   += volume_fractions[j] * (drucker_prager_parameters.cohesion * weakening_factors[0]);
                 // Also convert radians to degrees
-                plastic_out->friction_angles[i] += 180.0/numbers::PI * volume_fractions[j] * (drucker_prager_parameters.angle_internal_friction * weakening_factors[1]);
+                double current_friction = calculate_isostrain_viscosities(in, i, volume_fractions,
```

Comment:
It's a bit annoying we have to evaluate calculate_isostrain_viscosities again, but I don't see a simple solution at the moment.

Reply:
yes, I totally agree!